staging/src/k8s.io/apimachinery/pkg/util/cache/expiring.go

```diff
@@ -164,7 +147,7 @@ func (c *Expiring) gc() {
 		// from looking at the (*expiringHeap).Pop() implmentation below.
 		// heap.Pop() swaps the first entry with the last entry of the heap, then
 		// calls (*expiringHeap).Pop() which returns the last element.
-		if len(c.heap) == 0 || now.After(c.heap[0].expiry) {
+		if len(c.heap) == 0 || !now.After(c.heap[0].expiry) {
```

Comment:
How did this pass the tests previously?

Reply:
We were checking Get() which drops expired entries but not checking cache size.

Reply:
`c.heap[0].expiry.After(now)` lines up with the comment better IMO.